src/GivensQR.jl

```diff
@@ -0,0 +1,141 @@
+module GivensQR
+
+export givensqr_op
+
+using SparseArrays, LinearAlgebra, LinearOperators
+
+mutable struct GivensQR_model{T}
```

Comment:
```
struct GivensQR_model{T,Ti <: Integer,S <: AbstractMatrix}
  rows :: Array{Ti}
  ...
  R :: S
end
```
Try this, and let me know.

Reply:
Change name to `GivensQR` and change package name to `OrthogonalFactorizations.jl`

Reply:
And split into different files:
- `OrthogonalFactorizations.jl` which includes:
- `givens.jl` with everything else.

Rename `qr` to `givens_qr`. Later, we'll define `householder_qr`, and then `qr(..., method=:givens)`, for instance.